packages/lit-dev-content/site/tutorials/content/first-component/04.md

```diff
@@ -0,0 +1,104 @@
+By the end of this step, the shown word will automatically change every second.
+
+This can be achieved by using
+[`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/setInterval), a
+browser function which repeatedly calls a function with a fixed delay between
+each call.
+
+Define a method `tickToNextWord` that will increment `this.idx`.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  render() {
+    ...
+  }
+
+  tickToNextWord = () => { this.idx += 1; };
+
+```
+
+```js
+// word-viewer.js
+
+  render() {
+    ...
+  }
+
+  tickToNextWord = () => { this.idx += 1; };
+
+```
+
+{% endswitchable-sample %}
+
+Calling this method increments our reactive internal state `idx` which will
+schedule an update.
+
+To setup the interval and clean up the interval, use the [standard custom
+element lifecycle
+callbacks](/docs/components/lifecycle/#custom-element-lifecycle)
+`connectedCallback`, and `disconnectedCallback`.
+
+`connectedCallback` is invoked when the component is added to the DOM, and
+`disconnectedCallback` is invoked when the component is removed from the DOM.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  private intervalTimer?: number;
+  
+  connectedCallback() {
+    super.connectedCallback();
+    this.intervalTimer = setInterval(this.tickToNextWord, 1000);
+  }
+  
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.intervalTimer);
+    this.intervalTimer = undefined;
+  }
+
+```
+
+```js
+// word-viewer.js
+
+  intervalTimer;
+  
+  connectedCallback() {
+    super.connectedCallback();
+    this.intervalTimer = setInterval(this.tickToNextWord, 1000);
+  }
+  
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.intervalTimer);
+    this.intervalTimer = undefined;
+  }
+
+
+```
+
+{% endswitchable-sample %}
+
+When the `word-viewer` is attached to the DOM, `connectedCallback` is invoked
+and `setInterval` repeatedly calls `this.tickToNextWord` every 1000ms, or every
+second.
+
+Store the interval ID on `this.intervalTimer` which does not need to be reactive.
+
+When `word-viewer` is removed from the DOM, `disconnectedCallback` is invoked
+cleaning up the interval, and `this.tickToNextWord` is no longer called.
+
+The preview should be cycling over individual words.
+
+<litdev-aside type="info" no-header>
+
+Lit leverages standard [custom element lifecycle
+callbacks](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#using_the_lifecycle_callbacks).
```

Comment:
Is it worth mentioning in this aside why the `super.connectedCallback()` and `super.disconnectedCallback()` are necessary?

Reply:
Added to the aside.